BlueMapCommon/webapp/public/map_data_provider.php

```diff
@@ -126,67 +128,72 @@ function getMimeType($path) {
         $compression = $lod === 0 ? $hiresCompression : "none";
 
         // query for tile
-        $statement = $sql->prepare("
-            SELECT t.`data`
-            FROM `bluemap_map_tile` t
-             INNER JOIN `bluemap_map` m
-              ON t.`map` = m.`id`
-             INNER JOIN `bluemap_map_tile_compression` c
-              ON t.`compression` = c.`id`
-            WHERE m.`map_id` = ?
-            AND t.`lod` = ?
-            AND t.`x` = ?
-            AND t.`z` = ?
-            AND c.`compression` = ?
-        ");
-        $statement->bind_param("siiis", $mapId, $lod, $tileX, $tileZ, $compression);
-        $statement->execute();
-        if ($statement->errno) error(500, "Database query failed!");
-
-        // return result
-        $result = $statement->get_result();
-        if ($result && $line = $result->fetch_assoc()) {
-            if ($compression !== "none")
-                header("Content-Encoding: $compression");
-
-            if ($lod === 0) {
-                header("Content-Type: application/json");
-            } else {
-                header("Content-Type: image/png");
+        try {
+            $statement = $sql->prepare("
+                SELECT t.data
+                FROM bluemap_map_tile t
+                INNER JOIN bluemap_map m
+                ON t.map = m.id
+                INNER JOIN bluemap_map_tile_compression c
+                ON t.compression = c.id
+                WHERE m.map_id = :map_id
+                AND t.lod = :lod
+                AND t.x = :x
+                AND t.z = :z
+                AND c.compression = :compression
+            ");
+            $statement->bindParam( ':map_id', $mapId, PDO::PARAM_STR );
+            $statement->bindParam( ':lod', $lod, PDO::PARAM_INT );
+            $statement->bindParam( ':x', $tileX, PDO::PARAM_INT );
+            $statement->bindParam( ':z', $tileZ, PDO::PARAM_INT );
+            $statement->bindParam( ':compression', $compression, PDO::PARAM_STR);
+            $statement->setFetchMode(PDO::FETCH_ASSOC);
+            $statement->execute();
+
+            // return result
+            if ($line = $statement->fetch()) {
+                if ($compression !== "none")
+                    header("Content-Encoding: $compression");
+                if ($lod === 0) {
+                    header("Content-Type: application/json");
+                } else {
+                    header("Content-Type: image/png");
+                }
+                fpassthru($line["data"]);
+                exit;
             }
 
-            echo $line["data"];
-            exit;
-        }
+        } catch (PDOException $e) { error(500, "$e"); }
```

Comment:
Same as above:
> Which error-messages are being outputted here, and can we be sure that this is nothing which could be dangerous to send to the public? (E.g. backend-ip's/adresses, users etc..)

Reply:
That returns a very detailed PDO exception that details what went wrong.
It is indeed possible to leak info, I will change it to a generic message.
There is also the possibility of setting a "debug=true" somewhere in the future to output detailed information.